plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConfig.java

```diff
@@ -91,4 +98,73 @@ public IcebergConfig setMaxPartitionsPerWriter(int maxPartitionsPerWriter)
         this.maxPartitionsPerWriter = maxPartitionsPerWriter;
         return this;
     }
+
+    @NotNull
+    public String getMetadataImpl()
+    {
+        return metadataImpl;
+    }
+
+    @Config("iceberg.metadata-impl")
+    @ConfigDescription("the implementation class of Trino ConnectorMetadata")
+    public IcebergConfig setMetadataImpl(String metadataImpl)
+    {
+        this.metadataImpl = metadataImpl;
+        return this;
+    }
+
+    @NotNull
+    public String getCatalogImpl()
+    {
+        return catalogImpl;
+    }
+
+    @Config("iceberg.catalog-impl")
```

Comment:
Maybe a simpler way is to make this `metadataType` for users to choose among options like GENERIC, HIVE, etc. There are quite a few places that would require a switch statement and can take advantage of an enum.